src/algorithms/annealed_gibbs.jl

```diff
@@ -5,15 +5,13 @@ function annealed_gibbs!(
         model::Union{SeqModel,DistributedSeqModel},
         spikes::Vector{Spike},
         initial_assignments::Vector{Int64},
-        num_anneals::Int64,
-        samples_per_anneal::Int64,
-        max_temperature::Float64,
-        extra_split_merge_moves::Int64,
-        split_merge_window::Float64,
-        save_every::Int64;
+        config::Dict;
```

Comment:
Any reason why the arguments were replaced by `config`?

Two alternatives that might be safer:
1. Use keyword arguments, e.g., `num_anneals = 10`.
2. Create a struct `GibbsParameters(num_anneals, samples_per_anneal, <etc>)` and pass this as an argument.

Option 2 is essentially what you have here, but it's typed. Typing helps catch bugs early / avoid unexpected behavior when, for example, someone passes an array instead of a scalar for `max_temperature`.

Reply:
Aha yeah this does seem non-optimal, basically I added some extract variable that I needed to pass deep inside some nesting of functions and got annoyed having to go and change the argument list for every function on the way down repeatedly, so shoved everything in the config. Very open to the idea that this is not the ideal way to do things.

More than happy to have a go changing